packages/bitcore-wallet-service/src/lib/chain/eth/index.ts

```diff
@@ -134,14 +155,13 @@ export class EthChain implements IChain {
           } catch (err) {
             output.gasLimit = Defaults.DEFAULT_GAS_LIMIT;
           }
+          if (_.isNumber(opts.fee)) {
+            // This is used for sendmax
+            gasPrice = feePerKb = Number((opts.fee / (inGasLimit || Defaults.DEFAULT_GAS_LIMIT)).toFixed());
+          }
+          gasLimit = inGasLimit || Defaults.DEFAULT_GAS_LIMIT;
+          fee += feePerKb * gasLimit;
```

Comment:
Why this is inside the output loop? `          fee += feePerKb * gasLimit; `?

Reply:
We were comparing the incorrect amount of fee with the eth linked balance ( also showing the incorrect fee to pay on the confirm view on Copay ). There's two outputs involved (approve and pay) and we were only using the last one.

Example: Those are two invoices payments each of them has 2 fees related.
<img width="574" alt="BitPay" src="https://user-images.githubusercontent.com/10999037/98154180-298cba80-1eb3-11eb-9a3d-5f978439ce60.png">